audit/src/main/scala/tech/beshu/ror/audit/utils/AuditSerializationHelper.scala

```diff
@@ -223,8 +236,18 @@ private[ror] object AuditSerializationHelper {
 
     final case class StaticText(value: String) extends AuditFieldValueDescriptor
 
+    final case class BooleanValue(value: Boolean) extends AuditFieldValueDescriptor
+
+    final case class NumericValue(value: BigDecimal) extends AuditFieldValueDescriptor
+
     final case class Combined(values: List[AuditFieldValueDescriptor]) extends AuditFieldValueDescriptor
 
+    final case class Nested(values: Map[AuditFieldName, AuditFieldValueDescriptor]) extends AuditFieldValueDescriptor
```

Comment:
It's one level of nesting. Let's do it in a more generic way (to allow many nesting levels). 

BTW. We define fields like that:
```
fields: Map[AuditFieldName, AuditFieldValueDescriptor]
```

So, maybe we should not add Nested as a descriptor, but maybe we should modify the `AuditFieldName` definition instead.

E.g.
`AuditFieldName` is:
* a string OR
*  is a composition of `AuditFieldName`s (or strings)?

Reply:
I did it as suggested:
- there is now `AuditFieldPath`, that stores `NonEmptyList[String]` 
- AuditFieldValueDescriptor.Nested is removed
- the YAML representation is not changed - so the structure in yaml is nested, but it is decoded to flat Map[AuditFieldPath, AuditFieldValueDescriptor]

It required a bit of boilerplate and utils (because the structure in config is no longer 1:1 mirror of the domain representation in code). Invalid settings are detected. Cases 1 and 3 were already handled before review, case 2 is now handled when converting from nested structure in yaml to domain representation.